src/runtime-rs/crates/resource/src/network/network_pair.rs

```diff
@@ -177,3 +177,110 @@ pub async fn get_link_by_name(
 
     Ok(link::get_link_from_message(msg))
 }
+
+#[cfg(test)]
+pub mod test_util {
+    use crate::network::network_model::tc_filter_model::fetch_index;
+
+    // remove a link by its name
+    #[allow(dead_code)]
+    pub async fn delete_link(
+        handle: &rtnetlink::Handle,
+        name: &str,
+    ) -> Result<(), rtnetlink::Error> {
+        let link_index = fetch_index(handle, name)
+            .await
+            .expect("failed to fetch index");
+        // the ifindex of a link will not change during its lifetime, so the index
+        // remains the same between the query above and the deletion below
+        handle.link().del(link_index).execute().await
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use nix::unistd::getuid;
+    use scopeguard::defer;
+
+    use super::*;
+    use crate::network::{
+        network_model::TC_FILTER_NET_MODEL_STR, network_pair::test_util::delete_link,
+    };
+
+    // this ut tests create_link() and get_link_by_name()
+    #[actix_rt::test]
+    async fn test_utils() {
+        if let Ok((conn, handle, _)) =
+            rtnetlink::new_connection().context("failed to create netlink connection")
+        {
+            let thread_handler = tokio::spawn(conn);
+            defer!({
+                thread_handler.abort();
+            });
+
+            // create link needs to be root
+            if !getuid().is_root() {
```

Comment:
Please use `skip_if_not_root!` macro.

Refer: https://github.com/kata-containers/kata-containers/pull/4961/files#diff-0ea0de901151e555c6d92b8a788889e8fdb0a5a2e2fe2de17eae35eaa35bd35dR970

Reply:
I think I need to wait until the PR #4961 get merged.

Reply:
@Ji-Xinyou  No, the `skip_if_not_root!()` is already in the `main` branch, 4961 is only an example of how to use it.

Reply:
@liubin , I'll be working on this now

Reply:
Fixed, now it uses skip_if_not_root!() now. Nice catch @liubin.